src/sql/platform/query/common/gridDataProvider.ts

```diff
@@ -88,22 +88,24 @@ export async function getResultsString(provider: IGridDataProvider, selection: S
 		};
 	});
 
-	if (tasks.length > 0) {
-		let p = tasks[0]();
-		for (let i = 1; i < tasks.length; i++) {
-			p = p.then(tasks[i]);
-		}
-		await p;
-	}
+	// Set the tasks gathered above to execute
+	let actionedTasks: Promise<void>[] = tasks.map(t => { return t(); });
+
+	// Make sure all these tasks have executed
+	await Promise.all(actionedTasks);
```

Comment:
I think this is something we don't want. `Promise.all` runs the promises in parallel, but we want a sequential order and that's why we were running the tasks the way they were.